storage/conformance_test.go

```diff
@@ -16,21 +16,337 @@ package storage
+	if err != nil {
+		return err
+	}
+	rt.host = endpoint
+
+	c := http.DefaultClient
+	data := struct {
+		Instructions map[string][]string `json:"instructions"`
+	}{
+		Instructions: instructions,
+	}
+
+	buf := new(bytes.Buffer)
+	if err := json.NewEncoder(buf).Encode(data); err != nil {
+		return fmt.Errorf("encoding request: %v", err)
+	}
+
+	rt.host.Path = "retry_test"
+	resp, err := c.Post(rt.host.String(), "application/json", buf)
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("creating retry test: err: %v, resp: %+v", err, resp)
+	}
+	defer func() {
+		closeErr := resp.Body.Close()
+		if err == nil {
+			err = closeErr
+		}
+	}()
+	testRes := struct {
+		TestID string `json:"id"`
+	}{}
+	if err := json.NewDecoder(resp.Body).Decode(&testRes); err != nil {
+		return fmt.Errorf("decoding test ID: %v", err)
+	}
+
+	rt.id = testRes.TestID
+
+	// Create wrapped client which will send emulator instructions
+	rt.host.Path = ""
+	client, err := wrappedClient(rt.T, rt.host.String(), rt.id)
+	if err != nil {
+		return fmt.Errorf("creating wrapped client: %v", err)
+	}
+	rt.wrappedClient = client
+
+	return nil
+}
+
+// Verify that all instructions for a given retry testID have been used up.
+func (rt *retrySubtest) check() error {
+	rt.host.Path = strings.Join([]string{"retry_test", rt.id}, "/")
+	c := http.DefaultClient
+	resp, err := c.Get(rt.host.String())
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("getting retry test: err: %v, resp: %+v", err, resp)
+	}
+	defer func() {
+		closeErr := resp.Body.Close()
+		if err == nil {
+			err = closeErr
+		}
+	}()
+	testRes := struct {
+		Instructions map[string][]string
+		Completed    bool
+	}{}
+	if err := json.NewDecoder(resp.Body).Decode(&testRes); err != nil {
+		return fmt.Errorf("decoding response: %v", err)
+	}
+	if !testRes.Completed {
+		return fmt.Errorf("test not completed; unused instructions: %+v", testRes.Instructions)
+	}
+	return nil
+}
+
+// Delete a retry test resource.
+func (rt *retrySubtest) delete() error {
+	rt.host.Path = strings.Join([]string{"retry_test", rt.id}, "/")
+	c := http.DefaultClient
+	req, err := http.NewRequest("DELETE", rt.host.String(), nil)
+	if err != nil {
+		return fmt.Errorf("creating request: %v", err)
+	}
+	resp, err := c.Do(req)
+	if err != nil || resp.StatusCode != 200 {
+		return fmt.Errorf("deleting test: err: %v, resp: %+v", err, resp)
+	}
+	return nil
+}
+
+type testRoundTripper struct {
+	*testing.T
+	rt     http.RoundTripper
+	testID string
+}
+
+func (wt *testRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
+	r.Header.Set("x-retry-test-id", wt.testID)
+
+	requestDump, err := httputil.DumpRequest(r, false)
+	if err != nil {
+		wt.Logf("error creating request dump: %v", err)
+	}
+
+	resp, err := wt.rt.RoundTrip(r)
+	if err != nil {
+		wt.Logf("roundtrip error (may be expected): %v\nrequest: %s", err, requestDump)
+	}
+	return resp, err
+}
+
+// Create custom client that sends instruction
```

Comment:
Maybe add more details in the comment... something along the lines of "Create custom client that sends instruction to the storage testbench Retry Test API"